### PR TITLE
SendMessageInterceptorImpl repos refactor

### PR DESCRIPTION
**Pull request by leandroBorgesFerreira**
### 🎯 Goal

Divide `repos` inside `SendMessageInterceptorImpl` into many smaller interfaces so the interceptor can be moved to LLC module in a later PR. 

### 🛠 Implementation details

The idea is to divide https://github.com/GetStream/stream-chat-android/pull/3337 into smaller PRs so we can review it in an easier way. 
RepositoryFacade is not an interface and then `updateLastMessageForChannel` should be moved out of it so the actually implementation doesn't matter for clients of this piece of code.
Repository interfaces will be moved to LLC, which will help to move `SendMessageInterceptorImpl` there too. Later `LogicRegistry` and `GlobalState` will be moved outside `OfflinePlugin` and then the interceptor will be able to be moved to LLC. 

### 🧪 Testing

Send an attachment.

#### General
- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] Assigned a person / code owner group (required)
- [x] Thread with the PR link started in a respective Slack channel (#android-chat-core or #android-chat-ui) (required)
- [x] PR targets the `develop` branch
- [ ] PR is linked to the GitHub issue it resolves

#### Code & documentation
- ~[ ] Changelog is updated with client-facing changes~
- [x] New code is covered by unit tests
- ~[ ] Comparison screenshots added for visual changes~
- ~[ ] Affected documentation updated (KDocs, docusaurus, tutorial)~

### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

### 🎉 GIF

_Please provide a suitable gif that describes your work on this pull request_

### CODE REVIEW

File: stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/repository/domain/channel/internal/ChannelRepository.kt [hunk 1]
```diff
@@ -160,4 +162,29 @@ internal class ChannelRepositoryImpl(
     override fun clearChannelCache() {
         channelCache.evictAll()
     }
+
+    override suspend fun updateLastMessageForChannel(cid: String, lastMessage: Message) {
```
Comment (filbabic) [inline stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/repository/domain/channel/internal/ChannelRepository.kt:166]: Maybe some docs? IDK if we have them in the interface (if not, then add them there)

File: stream-chat-android-offline/src/test/java/io/getstream/chat/android/offline/repository/domain/channel/internal/ChannelRepositoryImplTest.kt [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        verify(channelDao).insert(
+            argThat { channelEntity ->
+                channelEntity.lastMessageAt == lastMessage.createdAt &&
+                    channelEntity.lastMessageId == lastMessage.id
+            }
+        )
+    }
+
+    @Test
+    fun `Given channel with outdated lastMessage in DB Should insert channel with updated last message`() = runTest {
```
Comment (filbabic) [inline stream-chat-android-offline/src/test/java/io/getstream/chat/android/offline/repository/domain/channel/internal/ChannelRepositoryImplTest.kt:65]: Should we also add a test where we check that when the date of the "new" message is somehow older than the previous last message, the `channelDao` has no interactions?

**Review by filbabic — COMMENTED** [review]
Seems to work fine! :]

I've left a comment or two, LMK once it's cleared up!

File: stream-chat-android-offline/src/test/java/io/getstream/chat/android/offline/repository/domain/channel/internal/ChannelRepositoryImplTest.kt (hunk 2 above)
Comment (leandroBorgesFerreira) [reply stream-chat-android-offline/src/test/java/io/getstream/chat/android/offline/repository/domain/channel/internal/ChannelRepositoryImplTest.kt:65]: Makes sense

**Comment by leandroBorgesFerreira** [comment]
https://github.com/GetStream/stream-chat-android/pull/3358

File: stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/repository/domain/channel/internal/ChannelRepository.kt (hunk 1 above)
Comment (leandroBorgesFerreira) [reply stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/repository/domain/channel/internal/ChannelRepository.kt:172]: I'm adding Kdocs in this PR: https://github.com/GetStream/stream-chat-android/pull/3358

**Review by filbabic — APPROVED** [review]
### ☑️Reviewer Checklist
- [x] UI Components sample runs & works
- [x] Compose sample runs & works
- [x] New feature tested and works

**Review by kanat — APPROVED** [review]
LGTM

**Pull request merged**